beacon-chain/core/helpers/beacon_committee.go

```diff
@@ -176,72 +177,98 @@ func CommitteeAssignments(
 	// we use a map of proposer idx -> []slot to keep track of this possibility.
 	startSlot, err := slots.EpochStart(epoch)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	minValidStartSlot := primitives.Slot(0)
 	if state.Slot() >= params.BeaconConfig().SlotsPerHistoricalRoot {
 		minValidStartSlot = state.Slot() - params.BeaconConfig().SlotsPerHistoricalRoot
 	}
 	if startSlot < minValidStartSlot {
-		return nil, nil, fmt.Errorf("start slot %d is smaller than the minimum valid start slot %d", startSlot, minValidStartSlot)
+		return nil, fmt.Errorf("start slot %d is smaller than the minimum valid start slot %d", startSlot, minValidStartSlot)
 	}
 
-	proposerIndexToSlots := make(map[primitives.ValidatorIndex][]primitives.Slot, params.BeaconConfig().SlotsPerEpoch)
+	assignmentMap := make(map[primitives.ValidatorIndex]*AssignmentContainer)
+	assignmentMap, err = fillProposerAssignments(ctx, startSlot, state, assignmentMap)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fill proposer assignments")
+	}
+
+	// If previous proposer indices computation is outside if current proposal epoch range,
+	// we need to reset state slot back to start slot so that we can compute the correct committees.
```

Comment:
Small grammar fixes
```suggestion
	// If previous proposer indices computation is outside of the current proposal epoch range,
	// we need to reset the state slot back to the start slot so that we can compute the correct committees.
```